src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -16,647 +16,146 @@
 {
-    VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INTERNAL);
-
-    FabricSceneData fabric(mEndpointId, fabric_index, mMaxScenesPerFabric, mMaxScenesPerEndpoint);
-    SceneTableData scene(mEndpointId, fabric_index);
-
-    ReturnErrorOnFailure(fabric.Load(mStorage));
-    VerifyOrReturnError(fabric.Find(scene_id, scene.index) == CHIP_NO_ERROR, CHIP_ERROR_NOT_FOUND);
-
-    CHIP_ERROR err = scene.Load(mStorage);
-
-    // If scene.Load returns "buffer too small", the scene in memory is too big to be retrieve (this could happen if the
-    // kMaxClustersPerScene was reduced by OTA) and therefore must be deleted as is is no longer considered accessible.
-    if (err == CHIP_ERROR_BUFFER_TOO_SMALL)
-    {
-        ReturnErrorOnFailure(this->RemoveSceneTableEntry(fabric_index, scene_id));
-    }
-    ReturnErrorOnFailure(err);
-
-    entry.mStorageId   = scene.mStorageId;
-    entry.mStorageData = scene.mStorageData;
-
-    return CHIP_NO_ERROR;
+    return this->GetTableEntry(fabric_index, scene_id, entry);
 }
 
 CHIP_ERROR DefaultSceneTableImpl::RemoveSceneTableEntry(FabricIndex fabric_index, SceneStorageId scene_id)
 {
-    VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INTERNAL);
-    FabricSceneData fabric(mEndpointId, fabric_index, mMaxScenesPerFabric, mMaxScenesPerEndpoint);
-
-    ReturnErrorOnFailure(fabric.Load(mStorage));
-
-    return fabric.RemoveScene(mStorage, scene_id);
+    return this->RemoveTableEntry(fabric_index, scene_id);
 }
 
-/// @brief This function is meant to provide a way to empty the scene table without knowing any specific scene Id. Outside of this
-/// specific use case, RemoveSceneTableEntry should be used.
-/// @param fabric_index Fabric in which the scene belongs
-/// @param scened_idx Position in the Scene Table
-/// @return CHIP_NO_ERROR if removal was successful, errors if failed to remove the scene or to update the fabric after removing it
 CHIP_ERROR DefaultSceneTableImpl::RemoveSceneTableEntryAtPosition(EndpointId endpoint, FabricIndex fabric_index,
                                                                   SceneIndex scene_idx)
 {
-    VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INTERNAL);
-
-    CHIP_ERROR err = CHIP_NO_ERROR;
-    FabricSceneData fabric(endpoint, fabric_index, mMaxScenesPerFabric, mMaxScenesPerEndpoint);
-    SceneTableData scene(endpoint, fabric_index, scene_idx);
-
-    ReturnErrorOnFailure(fabric.Load(mStorage));
-    err = scene.Load(mStorage);
-    VerifyOrReturnValue(CHIP_ERROR_NOT_FOUND != err, CHIP_NO_ERROR);
-    ReturnErrorOnFailure(err);
-
-    return fabric.RemoveScene(mStorage, scene.mStorageId);
+    return this->RemoveTableEntryAtPosition(endpoint, fabric_index, scene_idx);
 }
 
 CHIP_ERROR DefaultSceneTableImpl::GetAllSceneIdsInGroup(FabricIndex fabric_index, GroupId group_id, Span<SceneId> & scene_list)
 {
-    FabricSceneData fabric(mEndpointId, fabric_index, mMaxScenesPerFabric, mMaxScenesPerEndpoint);
```

Comment:
Now looking at this, it seems that this method could just as well use the entry_map (scene_map) in a similar way as DeleteAllScenesInGroup. Which would reduce the amount of times we load the fabric.

Reply:
Woopsie, this caused a bug due to uninitialized `scene_list` 😅🤣

That's why the tests were failing, should be all good now